tools/worker-runner/provider/provider/common.go

```diff
@@ -77,3 +78,15 @@ func RemoveWorker(state *run.State, factory tc.WorkerManagerClientFactory) error
 
 	return err
 }
+
+// ReportWorkerError will send a worker error report to worker-manager
+func ReportWorkerError(state *run.State, factory tc.WorkerManagerClientFactory, payload *tcworkermanager.WorkerErrorReport) error {
+	wc, err := factory(state.RootURL, &state.Credentials)
+	if err != nil {
+		return errors.Wrap(err, "error instanciating worker-manager client")
+	}
+	if _, err = wc.ReportWorkerError(state.WorkerPoolID, payload); err != nil {
+		log.Printf("Error reporting worker error: %v\n", err)
```

Comment:
If this error happens, it's probably got a common cause with the report from the worker, so perhaps we should log the payload as well?